### DISCUSSION TITLE
Missing Field 'secrets' after clean install

### DISCUSSION THREAD

**Discussion by l3ennji** [matrix-org/matrix-authentication-service#1961] [Q&A] [answered]
Hello,

I tried installing the mas on a Raspberry Pi 4 8Gb with the current Raspberry Pi OS 64bit, based on Debian Bookworm, and ran into an error after creating the config.yaml.
I have followed the installation instruction provided in the documentation, using the docker option. After creating the config.yaml using the mas-cli config generate command, the check of the configuration file fails, stating "missing field 'secrets'". The 'secrets' field is obviously included in the config.yaml, the necessary secrets and keys have been created, but it just doesn't seem to pick it up. I also tried specifying the config-file-location during the config check, but that didn't change anything. Deleting all unchanged fields in the config.yaml and only leaving the 'secrets' field within the file didn't change anything either.

Any help resolving this issue will be greatly appreciated.

Best regards

Ben

**Comment by sandhose** [comment]
This feels like the configuration file is not being picked up.

Are you running using the Docker image or the pre-built binaries?

By default, it will look for a config file in the "current working directory" of the process, but you can explicitly set it through arguments or through the `MAS_CONFIG` environment variable. Here is some guidance on how to debug the config files not being loaded: https://matrix-org.github.io/matrix-authentication-service/setup/general.html#using-and-inspecting-the-configuration-file

**Reply by l3ennji** [reply]
I'm running the docker image.

The file is present in the working directory, and I also tried specifying the file location using the --config argument with the mas-cli config check command. I will check the environment variable, but I doubt this will solve the issue.
I read the documentation beforehand and tried all listed debug options (except the environment variable) but without any success.

**Reply by sandhose** [reply]
Since you're running off the docker image, can you provide whatever configuration file is creating that container? A docker-compose file, kubernetes deployment, systemd service, or whatever you're using?
I'm mostly interested how you mount the configuration file inside the container

**Reply by l3ennji** [reply]
I'm using the exact commands and files provided in the documentation (docker build -t mas) and the current repo version, according to https://matrix-org.github.io/matrix-authentication-service/setup/installation.html.
So I suppose I'm using the regular Dockerfile.

**Reply by l3ennji** [reply]
[Dockerfile.txt](https://github.com/matrix-org/matrix-authentication-service/files/12928991/Dockerfile.txt)
This is the actual file, had to rename it to .txt so GitHub allows the file upload.

**Reply by sandhose** [reply, marked as answer]
You probably want to use pre-built images in production instead of building it yourself, like [ghcr.io/matrix-org/matrix-authentication-service:main](https://ghcr.io/matrix-org/matrix-authentication-service:main).

I'm asking what are you using to run the service? Simple `docker run` invocations? If so, what's the exact command you're running, and how do you make the configuration file available within the container?

If you feel like you're not familiar with Docker, you can also try and setup the service with the pre-built binaries available on [the latest release](https://github.com/matrix-org/matrix-authentication-service/releases/latest). In your case, that would be the [`mas-cli-aarch64-linux.tar.gz`](https://github.com/matrix-org/matrix-authentication-service/releases/latest/download/mas-cli-aarch64-linux.tar.gz) archive, as you said you're running on a Raspberry Pi.

**Reply by l3ennji** [reply]
Thanks for the hint that the pre-built binaries are actually raspi compatible.
I downloaded the pre-built binary and now the configuration check runs without any issues. The database migrate command works as well (that's where I actually got stuck with the docker image), so I should be fine for now. Thanks a lot for your immediate and helpful response. Looks like I have some catching up to do when it comes to docker images. I used one for my home-assistant-node, but never actually really messed around with it, because it usually ran out of the box.
I sincerely apologize for wasting your time due to my lack of knowledge.
Maybe a small hint (USE THIS BINARY IF YOU'RE A NOOB WITH A RASPI!) could save you some time the next time someone like me comes around and doesn't understand that aarch64 is the same as ARM64. (I'm going to return my IT-degree now...)
